flink-java/src/main/java/org/apache/flink/api/java/operators/SortPartitionOperator.java

```diff
@@ -89,6 +100,22 @@ public SortPartitionOperator(DataSet<T> dataSet, String sortField, Order sortOrd
 		return this;
 	}
 
+	/**
+	 * Appends an additional sort order with the specified field in the specified order to the
+	 * local partition sorting of the DataSet.
+	 *
+	 * @param keyExtractor The KeySelector function which extracts the key value of the additional
+	 *                     sort order of the local partition sorting.
+	 * @param order The order of the additional sort order of the local partition sorting.
+	 * @return The DataSet with sorted local partitions.
+	 */
+	public <K> SortPartitionOperator<T> sortPartition(KeySelector<T, K> keyExtractor, Order order) {
```

Comment:
I would not allow chaining.
In addition, we must make sure that `ExpressionKeys` and `SelectorFunctionKeys` are not mixed. So you can either use a single key selector or one or more expression keys.